engine/compiler/script.go

```diff
@@ -50,7 +50,12 @@ func setupScriptWindows(commands []string, dst *engine.Step) {
 // helper function configures the pipeline script for the
 // linux operating system.
 func setupScriptPosix(commands []string, dst *engine.Step) {
-	dst.Entrypoint = []string{"/bin/sh", "-c"}
-	dst.Command = []string{`echo "$DRONE_SCRIPT" | /bin/sh`}
+	if dst.Shell == "" {
+		dst.Entrypoint = []string{"/bin/sh", "-c"}
+		dst.Command = []string{`echo "$DRONE_SCRIPT" | /bin/sh`}
+	} else {
+		dst.Entrypoint = []string{dst.Shell, "-c"}
```

Comment:
I think this can be reverted. There is no guarantee the user-defined shell supports `-c` or that the flag will have the same meaning. But more importantly, I don't think it is required since we pipe the script to the target shell in the line below.